applications/luci-app-vnstat2/htdocs/luci-static/resources/view/vnstat2/graphs.js

```diff
@@ -16,41 +65,46 @@ return view.extend({
 		]);
 
 		ifaces.forEach(function(iface) {
-			tab.appendChild(E('span', {}, E('img', { 'data-iface': iface, 'style': 'visibility:hidden; margin:5px 10px' })));
 			fs.exec_direct('/usr/bin/vnstati', [ '-'+style, '-i', iface, '-o', '-' ], 'blob').then(function(res) {
 				var img = tab.querySelector('img[data-iface="%s"]'.format(iface));
 				img.src = URL.createObjectURL(res);
 				img.style.visibility = 'visible';
+				img.id = 'graph_' + style + '_' + iface;
 				tab.firstElementChild.style.display = 'none';
 			});
+			tab.appendChild(E('span', {}, E('img', { 'data-iface': iface, 'style': 'visibility:hidden; margin:5px 10px' })));
 		});
 
 		return tab;
 	},
 
 	load: function() {
-		return fs.exec_direct('/usr/bin/vnstat', [ '--json', 'f', '1' ], 'text').then(function(res) {
-			var json = null;
-
-			try {
-				json = JSON.parse(res)
-			}
-			catch(e) {
-				throw new Error(res.replace(/^Error: /, ''));
-			}
-
-			return (L.isObject(json) ? L.toArray(json.interfaces) : []).map(function(iface) {
-				return iface.name;
-			}).sort();
-		}).catch(function(err) {
-			ui.addNotification(null, E('p', { 'style': 'white-space:pre' }, [err]));
-			return [];
-		});
+		return Promise.all([
+			L.resolveDefault(callServiceList('vnstat'), {}),
```

Comment:
This fails with a "Access denied" response. You would need to add something like the following to the ACL file:
```
"ubus": {
	"service": [ "list" ]
},
```
(This probably works for your, because you installed another package that allows this call.)